lib/IRGen/GenCall.cpp

```diff
@@ -1462,9 +1462,13 @@ void SignatureExpansion::expandExternalSignatureTypes() {
   // Generate function info for this signature.
   auto extInfo = clang::FunctionType::ExtInfo();
 
-  auto &FI = clang::CodeGen::arrangeFreeFunctionCall(IGM.ClangCodeGen->CGM(),
-                                             clangResultTy, paramTys, extInfo,
-                                             clang::CodeGen::RequiredArgs::All);
+  bool isCXXMethod =
+      FnType->getRepresentation() == SILFunctionTypeRepresentation::CXXMethod;
+  auto &FI = isCXXMethod ?
+      clang::CodeGen::arrangeCXXMethodCall(IGM.ClangCodeGen->CGM(),
+          clangResultTy, paramTys, extInfo, clang::CodeGen::RequiredArgs::All) :
+      clang::CodeGen::arrangeFreeFunctionCall(IGM.ClangCodeGen->CGM(),
+          clangResultTy, paramTys, extInfo, clang::CodeGen::RequiredArgs::All);
```

Comment:
Now that we're arranging the call properly, can we change the places in IRGen that reorder the `sret` argument on Windows targets to just use `isSRetAfterThis()`?  This probably wasn't being set before, so it's understandable that we had to adjust it, but now that we've got an accurate `CGFunctionInfo`, we should be able to just trust this bit.